cloudcomposerdiff/CONTRIBUTING.md

```diff
@@ -0,0 +1,147 @@
+## Developing this application
+
+### Prerequisites
+
+ * Python version: application has been tested on python version 3.10
+ * Composer version:  compatible with both Composer 1 and Composer 2
+ * Composer version:  compatible with all Composer configurations
+ * GCP IAM Permissions: executing user needs composer.environments.get [permission](https://cloud.google.com/composer/docs/how-to/access-control#permissions_for_api_methods)
+
+### Clone repository
+
+Clone the code from GitHub to your local machine
+
+### Getting the correct version of python with pyenv
+
+Install [pyenv](https://github.com/pyenv/pyenv).
+
+Use pyenv to install a suitable version of python, e.g. 3.10.0
+```shell
+pyenv install 3.10.0    
+```
+
+Use pyenv to switch to python 3.10.0
+```shell
+pyenv shell 3.10.0
+```
+
+### Create a virtual environment with venv module
+
+```shell
+python -m venv venv
+```
+
+activate the virtual environment
+```shell
+source venv/bin/activate
+```
+check that we are using the Python of the virtual environment
+```
+which python
+.../cloudcomposerdiff/venv/bin/python
+```
+
+### Install the package in editable mode
+
+Install the package locally within the created virtual environment.
+
+Do so in "editable mode" so that the package will reflect code changes.
+
+Navigate to the directory containing pyproject.toml
+
+Install all the dependcies end-users would get when running pip install
+
+```
+python -m pip install --editable .
+```
+Install additional dev only dependencies that end-users do not get
+```shell
+python -m pip install --editable .[dev]
+```
+
+with the above done, running the package will include code changes. This can
+be used for testing during development.
+```
+cloudcomposerdiff \
+--env1_project_id <> \
+--env1_location <> \
+--env1_name <> \
+--env2_project_id <> \
+--env2_location <> \
+--env2_name <>
+```
+### Run tests w/ pytest
+
+Navigate to the directory containing pyproject.toml
+```shell
+pytest
+```
+
+### Sort imports with isort
+
+Navigate to the directory containing pyproject.toml
+
+Apply the sort to all files recurisvely
+
+```shell
+isort .
+```
+
+### Format the code with black
+
+Navigate to the directory containing pyproject.toml
+
+Apply the formatter to all files recurisvely
+
+```shell
+black .
+```
+
+### Lint the code
+
+Navigate to the directory containing pyproject.toml
+
+Apply the formatter to all files recurisvely
+
+```shell
+flake8 --max-line-length=88 src
```

Comment:
```suggestion
flake8 . --show-source --builtin=gettext --max-complexity=29 --import-order-style=google --ignore=E121,E123,E126,E203,E226,E24,E266,E501,E704,W503,W504,I202 --max-line-length=88
```

I know this is a huge amount of text but it's copied straight from the GH action https://github.com/GoogleCloudPlatform/composer-utilities/blob/4864e3ca553438e223b86e71e111221be3409574/.github/workflows/cloudcomposerdiff.yml#L46 (and derived from the samples lint that we use for cloud python samples)

Reply:
note that users would also have to install two additional dependencies (`flake8-import-order` and `flake8-annotations`)

Reply:
Thank you for sorting all this out Leah ! 

I will update the dev dependencies of pyproject.toml to include flake8-import-order and flak8-annotations